float32 dimX
float32 dimY
float32 dimZ
float32 offX
float32 offY
float32 offZ